### PR TITLE
Font and Color Best Practices Updates

### PR DESCRIPTION
**Pull request by mliberatore**
Closes [this notion task](https://www.notion.so/lickability/126ab953176b4d5aac19e04e3f768b38?v=d9c1415c574e4a088523240f4fed7e66&p=a7e461e53a024739b8fffa9dd9a356ca)

## What it Does

Updates/adds clarity around where and how to set colors and fonts.

## How I Tested

* Proofread.
* Made sure I covered every bullet on the notion task.
* Made sure the combined doc changes were identical by selecting, copying, and finding the entire changed blocks and ensuring there were two results for each in the diff.

## Notes

N/A

## Screenshot

N/A

### CODE REVIEW

File: Assets.md [hunk 1]
```diff
@@ -4,6 +4,9 @@ We use asset catalogs to organize our application’s assets.
 
 ### Key Considerations
 * Organize assets using folders inside asset catalogs.
-* Whenever possible, use images and colors from asset catalogs using Interface Builder.
-* When assets cannot be used from Interface Builder (i.e. the assets to use from the catalog are not known at compile time), make use of code generation solutions such as [SwiftGen](https://github.com/SwiftGen/SwiftGen#asset-catalog) to provide non-optional constants for assets.
+* All colors and images that are known at compile time should exist in asset catalogs.
```
Comment (mattbischoff) [inline Assets.md:7]: Should you create colors that point to system colors or use the system colors directly?

File: Assets.md [hunk 2]
```diff
@@ -4,6 +4,9 @@ We use asset catalogs to organize our application’s assets.
 
 ### Key Considerations
 * Organize assets using folders inside asset catalogs.
-* Whenever possible, use images and colors from asset catalogs using Interface Builder.
-* When assets cannot be used from Interface Builder (i.e. the assets to use from the catalog are not known at compile time), make use of code generation solutions such as [SwiftGen](https://github.com/SwiftGen/SwiftGen#asset-catalog) to provide non-optional constants for assets.
+* All colors and images that are known at compile time should exist in asset catalogs. 
+* Asset catalog colors should come from a project’s design system for project-wide consistency, ease of small tweaks, and to assist in supporting color-related features like dark mode. Avoid adding colors for specific use cases without verifying it with the project’s designers and updating the design system.
+* Whenever possible, set images from asset catalogs in Interface Builder. 
+* Always set colors from asset catalogs in code, never in Interface Builder.
```
Comment (mattbischoff) [inline Assets.md:10]: Might be good to add reasoning here.

File: Assets.md [hunk 3]
```diff
@@ -4,6 +4,9 @@ We use asset catalogs to organize our application’s assets.
...
+* When assets cannot be used from Interface Builder (i.e. images not available at compile time or colors), make use of code generation solutions such as [SwiftGen](https://github.com/SwiftGen/SwiftGen#asset-catalog) to provide non-optional constants for assets.
```
Comment (mattbischoff) [inline Assets.md:11]: The parenthetical here confused me. Maybe reword?

File: InterfaceBuilder.md [hunk 4]
```diff
@@ -6,8 +6,9 @@
 * Use `UIStackView`s instead of explicit constraints between siblings whenever possible, unless there are noticeable performance issues.
 * Each nib should have a single top level item.
     * Separate `UIView` subclasses designed in Interface Builder into their own nib files.
-* Use `IBDesignable` and `IBInspectable` for common design properties, e.g. specifying a view’s corner radius or giving it a border.
-* When setting colors in Interface Builder, use the ones defined in asset catalogs to ensure that all use of common colors is updated from a single source.
+* Use `IBInspectable` for common design properties, e.g. specifying a view’s corner radius or giving it a border. Avoid use of `IBDesignable` due to [outstanding tooling issues](http://www.openradar.appspot.com/radar?id=4952346488471552).
```
Comment (mattbischoff) [inline InterfaceBuilder.md:9]: Apple suggested during our call that this may not be a bug and could be working as intended. So I’d like to revise this to say use `IBDesginable` only for views with custom drawing (not to customize outlets) and use `IBInspectable` to customize properties of outlets.

File: Assets.md (hunk 2 above)
Comment (mliberatore) [reply Assets.md:10]: Repeat what’s in the IB doc?

> It is too easy to fall out of sync with asset catalog and design system colors when they’re set in Interface Builder.

File: Assets.md (hunk 1 above)
Comment (mliberatore) [reply Assets.md:7]: Good question. I’m inclined to say the latter, so "All colors and images that are known at compile time should exist in asset catalogs, unless they are system colors, in which case they do not need to be duplicated in the asset catalog." Any preference there, @Twigz?

Note this would also change the statement below to:

> Always set asset catalog and system colors in code…

or just

> Always set UI component colors in code…

File: Assets.md (hunk 3 above)
Comment (mliberatore) [reply Assets.md]: Simplified

File: Assets.md (hunk 2 above)
Comment (mliberatore) [reply Assets.md]: Did this ^

File: InterfaceBuilder.md (hunk 4 above)
Comment (mliberatore) [reply InterfaceBuilder.md]: Please let me know if the update here matches your intention.

File: Assets.md (hunk 1 above)
Comment (mliberatore) [reply Assets.md]: Updated according to the comment above, but would like @Twigz’ input

Comment (Twigz) [reply Assets.md]: Good to me!

**Comment by mliberatore** [comment]
Thanks!
